apps/api/src/app/exchange-rate/exchange-rate.service.spec.ts

```diff
@@ -0,0 +1,198 @@
+import { ExchangeRateService } from './exchange-rate.service';
+import { Currency, MarketData } from '@prisma/client';
+
+import { MarketDataService } from '../core/market-data.service';
+import { addDays, endOfDay, isBefore } from 'date-fns';
+import { Big } from 'big.js';
+import { DateQuery } from '@ghostfolio/api/app/core/interfaces/date-query.interface';
+
+jest.mock('../core/market-data.service', () => {
+  return {
+    MarketDataService: jest.fn().mockImplementation(() => {
+      return {
+        getRange: ({
+          dateQuery,
+          symbols
+        }: {
+          dateQuery: DateQuery;
+          symbols: string[];
+        }) => {
+          const exchangeRateMap = {
+            USDEUR: 1,
+            USDCHF: 2,
+            USDUSD: 0
+          };
+          const result = [];
+          let j = 1;
+          for (
+            let i = dateQuery.gte;
+            isBefore(i, dateQuery.lt);
+            i = addDays(i, 1)
+          ) {
+            const marketPrice = j++;
+            for (const symbol of symbols) {
+              result.push({
+                createdAt: i,
+                date: i,
+                id: '8fa48fde-f397-4b0d-adbc-fb940e830e6d',
```

Comment:
Do you want the id to be always the same? You could also do something random like:

```
import { v4 as uuidv4 } from 'uuid';

...

id: uuidv4(),
```

Reply:
No, I don't care about the id. We could use a random one as well.

Reply:
I left the static IDs, since those are completely irrelevant and I would prefer to have something static rather than something random in test cases.